code/modules/mob/living/basic/slime/life.dm

```diff
@@ -0,0 +1,68 @@
+
+/mob/living/basic/slime/Life(seconds_per_tick = SSMOBS_DT, times_fired)
+	..()
+
+	if(bodytemperature <= (T0C - 40)) // stun temperature
+		apply_status_effect(/datum/status_effect/freon, SLIME_COLD)
+	else
+		remove_status_effect(/datum/status_effect/freon, SLIME_COLD)
+
+	var/bz_percentage =0
+
+	var/turf/open/our_turf = loc
+
+	if(our_turf?.air.gases[/datum/gas/bz])
+		bz_percentage = our_turf.air.gases[/datum/gas/bz][MOLES] / our_turf.air.total_moles()
```

Comment:
Feels appropriate to put this stuff in `handle_environment`, rather than life directly, unless there's a reason it's here

Reply:
On a further look, the bodytemperature stuff definitely, but unsure about BZ. I will move it regardless.